codice/ddf
# DDF-4307 fix 'All Sources' behavior in sources drop down

#### What does this PR do?
Creating a new search, when you select the sources drop down, the 'All Sources' will remain checked unless you specifically uncheck it
fixes this for better ux (see demo below)
#### Who is reviewing it? 
@Bdthomson @hayleynorton @lamhuy 
#### Ask 2 committers to review/merge the PR and tag them here.
@andrewkfiedler
@bdeining
#### How should this be tested?
- build/run 
- start a new query and open the sources drop down
- verify that 'All Sources' is unselected when another source is selected
- verify all other options unselected when 'All Sources' is selected
- verify no regression in functionality
#### Any background context you want to provide?
#### What are the relevant tickets?
[DDF-4307](https://codice.atlassian.net/browse/DDF-4307)
#### Screenshots
![selectall](https://user-images.githubusercontent.com/39923178/48083417-603f7580-e1b2-11e8-97b1-c179031aec43.gif)

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
